doc/arch/adr-003-data-retention.md

```diff
@@ -0,0 +1,287 @@
+to Subscriptions and Subscribers indefinitely. As this is a somewhat low volume
+of data this shouldn't pose a huge issue storage wise.
+
+A field would be added to subscriber of `deactivated_at` which would be used to
+determine whether a subscriber is activated. Once a Subscriber has been
+deactivated for 28 days any personal information will be removed from the
+account. Any additional activity on the account will cause the account to be
+reactivated.
+
+_Aside: The `deactivated_at` field may not be required it could be determined
+by the lack of any active subscriptions. It just might be useful to have this
+as queries regarding end of a subscriber would be complex without it._
+
+This 28 day window is intended to provide us with sufficient time that we can
+deal with any support queries that occur due to account issues (as this is a
+reasonably regular occurance).
+
+With subscriptions the intention is to use this as an append only table and
+any change to a subscription (starting, ending, change of frequency) creates
+a new entry into this table. There will however be an active flag which can be
+used as part of the unique index on this to ensure a user can only have one
+active subscription at a time. This exception would not make it fully append
+anly.
+
+An additional change will be to add a timestamp, `ended_at`, which is used to
+record the time that the subscription ended at. Which will enable queries which
+determine the number of subscriptions at particular time periods. Additionally
+the suggestion is to include fields of the source of creating the subscription
+and to include the reason the subscription was ended.
+
+Finally we have a uuid field stored on subscription, it would likely be better
+for us to have a primary key that is the uuid and then not have a second
+numerical id.
+
+### Summary of database changes
+
+#### Email
+
+- Change the id type from a serial to a uuid
+- Add an archived_at timestamp
+- Would be removed from database after 14 days
+
+#### EmailArchive
+
+- A table which is store abridged Email data with the intention of this moving
+  to S3 (or a different data analytics storage)
+    - On this store information such as id, subject, subscriber_id,
+      content_changes (see example above)
+    - Store no personally identifiable information
+
+#### DeliveryAttempt
+
+- Change id to a uuid
+- Drop reference field and use uuid instead
+- Would be removed (as part of a cascade with Email) after 14 days
+
+#### SubscribtionContent
+
+- Would be deleted automatically when the corresponding Email is deleted
+
+#### ContentChange
+
+- Should not be deleted, as part of analytics picture
+
+#### SubscriberList
+
+- Should not be deleted, as part of analytics picture
+
+#### Subscriber
+
+- These should not be deleted
+- Add a `deactivated_at` timestamp field
+- Support a null value for address
+- Should not be deleted, as part of analytics picture
+
+#### Subscription
```

Comment:
What's the distinction between `ended_at` and `active` and are either (or both) of these already represented by `deleted_at`?

Reply:
Active would be for the purpose of a unique constraint to prevent the same subscriber to have the same subscription multiple times. 

Ended_at serves same purpose as deleted_at but was just an attempt to make the temporary more permanent. I haven't phrased it as that though I kinda just ignored deleted_at. I'll rephrase that as a name change since `deleted_at` might be a bit confusing if we're keeping it around forever.